brew_wizard_ui/src/app/style/style.service.ts

```diff
@@ -4,42 +4,43 @@ import { Observable } from 'rxjs/Observable';
 import 'rxjs/add/operator/map';
 import 'rxjs/add/operator/catch';
 import 'rxjs/add/operator/toPromise';
-import { Angular2TokenService } from 'angular2-token';
+import { HttpClient, HttpResponse } from '@angular/common/http';
 import { Response } from '@angular/http';
+import { environment } from '../../environments/environment';
 
 @Injectable()
 export class StyleService {
 
-  constructor(private _angularTokenService: Angular2TokenService) { }
+  constructor(private http: HttpClient) { }
 
   getStyles(): Observable<IStyle[]> {
-    return this._angularTokenService.get("styles")
-      .map((response: Response) => <IStyle[]>response.json())
+    return this.http.get(`${environment.token_auth_config.apiBase}/styles`)
+      .map((response: IStyle[]) => response)
       .catch(this.handleError);
   }
 
   getWaterProfiles(): Observable<any[]> {
-    return this._angularTokenService.get("water_profiles")
-      .map((response: Response) => <any[]>response.json())
+    return this.http.get(`${environment.token_auth_config.apiBase}/water_profiles`)
+      .map((response: any[]) => response)
       .catch(this.handleError);
   }
 
   getStyleById(styleId: number): Observable<IStyle> {
-    return this._angularTokenService.get("styles/" + styleId)
-      .map((response: Response) => <IStyle>response.json())
+    return this.http.get(`${environment.token_auth_config.apiBase}/styles/` + styleId)
+      .map((response: IStyle) => response)
       .catch(this.handleError);
   }
 
   createStyle(style: IStyle): any {
-    return this._angularTokenService.post("styles/", {style})
+    return this.http.post("styles/", {style})
```

Comment:
It looks like this need to be changed to `${environment.token_auth_config.apiBase}/styles/` like above. This revealed a couple other bugs too ><

1) Styles create/edit has Ideal Water Profile as a required item - it either needs to be optional, the ideal water profile dropdown needs to be automatically set to the first one, or the dropdown needs 'required' added to it to prevent hitting the create button before submitting.

2) On the backend the water_profile_id is not a permitted parameter.

Reply:
I am working on these fixes right now. For the water profile id, should I remove it or should it just be added to the allowed params on the backend. I can add it to the rails project if you want but I still have not setup a working copy of the database so I would not be able to test it immediately.

As for the services, I just ran a quick regex find and replace and fixed all the paths that needed the base api. Ill commit to the PR in a few.